sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -117,8 +123,19 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
         case LongType => tryParseLong(field)
         case _: DecimalType => tryParseDecimal(field)
         case DoubleType => tryParseDouble(field)
-        case TimestampNTZType => tryParseTimestampNTZ(field)
-        case TimestampType => tryParseTimestamp(field)
+        case DateType => tryParseDateTime(field)
```

Comment:
It seems changing the method `tryParseDouble` should be enough

Reply:
Consider the column of a TimestampType followed by a DateType entry. We would expect this column to be inferred as a TimestampType column.
`typeSoFar` will be `Timestamp` when `inferField` is called on the second entry which is DateType. We need logic in `inferField` to try and parse Date type even when `typeSoFar` is `Timestamp`